src/index.js

```diff
@@ -1,390 +1,604 @@
-    if (cb) {
-      lifecycle.push(function() {
-        cb(element, oldAttributes)
-      })
-    }
-  }
+var patch = function(container, element, lastNode, nextNode, eventProxy) {
+  var lifecycle = []
 
-  function removeChildren(element, node) {
-    var attributes = node.attributes
-    if (attributes) {
-      for (var i = 0; i < node.children.length; i++) {
-        removeChildren(element.childNodes[i], node.children[i])
-      }
+  element = patchElement(
+    container,
+    element,
+    lastNode,
+    nextNode,
+    lifecycle,
+    eventProxy
+  )
 
-      if (attributes.ondestroy) {
-        attributes.ondestroy(element)
-      }
-    }
-    return element
-  }
+  while (lifecycle.length > 0) lifecycle.pop()()
 
-  function removeElement(parent, element, node) {
-    function done() {
-      parent.removeChild(removeChildren(element, node))
-    }
+  return element
+}
 
-    var cb = node.attributes && node.attributes.onremove
-    if (cb) {
-      cb(element, done)
-    } else {
-      done()
+export var h = function(name, props) {
+  var node
+  var rest = []
+  var children = []
+  var length = arguments.length
+
+  while (length-- > 2) rest.push(arguments[length])
+
+  if ((props = props == null ? {} : props).children != null) {
+    if (rest.length <= 0) {
+      rest.push(props.children)
     }
+    delete props.children
   }
 
-  function patch(parent, element, oldNode, node, isSvg) {
-    if (node === oldNode) {
-    } else if (oldNode == null || oldNode.nodeName !== node.nodeName) {
-      var newElement = createElement(node, isSvg)
-      parent.insertBefore(newElement, element)
-
-      if (oldNode != null) {
-        removeElement(parent, element, oldNode)
+  while (rest.length > 0) {
+    if (isArray((node = rest.pop()))) {
+      for (length = node.length; length-- > 0; ) {
+        rest.push(node[length])
       }
-
-      element = newElement
-    } else if (oldNode.nodeName == null) {
-      element.nodeValue = node
+    } else if (node === false || node === true || node == null) {
     } else {
-      updateElement(
-        element,
-        oldNode.attributes,
-        node.attributes,
-        (isSvg = isSvg || node.nodeName === "svg")
-      )
-
-      var oldKeyed = {}
-      var newKeyed = {}
-      var oldElements = []
-      var oldChildren = oldNode.children
-      var children = node.children
+      children.push(typeof node === "object" ? node : createTextVNode(node))
+    }
+  }
 
-      for (var i = 0; i < oldChildren.length; i++) {
-        oldElements[i] = element.childNodes[i]
+  return typeof name === "function"
+    ? name(props, (props.children = children))
+    : createVNode(name, props, children, null, props.key, DEFAULT)
+}
 
-        var oldKey = getKey(oldChildren[i])
-        if (oldKey != null) {
-          oldKeyed[oldKey] = [oldElements[i], oldChildren[i]]
-        }
-      }
+var cancel = function(sub) {
+  sub[1][2]()
```

Comment:
I think this is actually a typo and should instead be:
```js
sub[2]()
```

Reply:
Good catch, I'll PR something in the next hour